QuasarHelios2015/src/org/team1540/quasarhelios/AutonomousModeBase.java

```diff
@@ -0,0 +1,45 @@
+package org.team1540.quasarhelios;
+
+import ccre.channel.BooleanInputPoll;
+import ccre.ctrl.BooleanMixing;
+import ccre.instinct.AutonomousModeOverException;
+import ccre.instinct.InstinctModeModule;
+
+public abstract class AutonomousModeBase extends InstinctModeModule {
+	public AutonomousModeBase(String modeName) {
+		super(modeName);
+	}
+	
+	protected void drive(float distance) throws AutonomousModeOverException, InterruptedException {
+		float currentEncoder = DriveCode.leftEncoder.get();
+		DriveCode.allMotors.set(distance > 0 ? 1.0f : -1.0f);
+
+		waitUntilAtLeast(DriveCode.leftEncoder, currentEncoder + distance);
```

Comment:
Any particular reason that you're multiplying both sides of the inequality by encoderScaling instead of dividing distance by encoderScaling? Also, perhaps you want to switch between waitUntilAtLeast and waitUntilAtMost based on the direction that you're going.